pkg/services/featuremgmt/manager.go

```diff
@@ -0,0 +1,196 @@
+package featuremgmt
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type FeatureManager struct {
+	isDevMod  bool
+	licensing models.Licensing
+	flags     map[string]*FeatureFlag
+	enabled   map[string]bool // only the "on" values
+	toggles   *FeatureToggles
+	config    string // path to config file
+	vars      map[string]interface{}
+	log       log.Logger
+}
+
+// This will merge the flags with the current configuration
+func (fm *FeatureManager) registerFlags(flags ...FeatureFlag) {
+	for idx, add := range flags {
+		if add.Name == "" {
+			continue // skip it with warning?
+		}
+		flag, ok := fm.flags[add.Name]
+		if !ok {
+			fm.flags[add.Name] = &flags[idx]
+			continue
+		}
+
+		// Selectively update properties
+		if add.Description != flag.Description {
```

Comment:
Should the check that add.Description is not empty be here? Moreover, maybe Description should be `*string` to allow overrides like this?